### DISCUSSION TITLE
Saving large image

### DISCUSSION THREAD

**Discussion by pravinohalhbg** [dlemstra/Magick.NET#720] [Help]
Hi I'm creating an a large image in memory with dimension as 12240x28672, with 32 bit depth and size 384mb
I'm trying to save the im memory image to disk, it is saving perfectly fine, but taking 16 minutes to save. That is lot of time.
I'm using Magick.NET-Q16-AnyCPU v7.21.1.0, Machine configuratio is I7 with 16 GB RAM
Here is that code that I'm using 
 using (var scannedImage = wholeImage.AppendHorizontally())
 {
                        
    // Save the result                            
    scannedImage.Write(savePath);//Takes 16 minutes
                        
 }
 
 My question is how can we save the image in minimal time. Thanks in Advance

**Comment by dlemstra** [comment]
When you use the `AnyCPU` version of the library it is possible that you are using the `x86` version instead of the `x64` version. With the `x86` version you will run out of memory quickly. And what is your output format?

**Reply by pravinohalhbg** [reply]
@dlemstra I'll try for x64 and check the performance and let you know, Thanks. The output format is PNG
